[FEAT] 연락처 작업

rhkrwngud445 opened the pull request:
## 작업 내용
- 연락처 화면
- 연락처 로드
- 선택 후 연락처 파라미터 전달


## 확인 방법
<img width="308" height="667" alt="스크린샷 2025-09-16 오후 9 04 51" src="https://github.com/user-attachments/assets/ee3e64b8-15c7-4cee-8737-d12acd3c64c0" />


## 참고 사항
**연락처 권한 직접 추가** 후 확인 가능합니다!
- 권한 작업은 이전 화면에서 진행

## 관련 이슈
- close #27

stopstone committed:
feat: 스플래쉬 화면 구현

stopstone committed:
feat: 로그인 상태 확인 로직 구현

- 스플랫쉬 화면에서 이전에 로그인이 되었는지 확인
- DataStore에 저장된 로그인 정보가 있다면 홈 화면
- 없다면 로그인 화면으로 이동한다.
- 상태를 불러오는데 실패하면 로그인 화면으로 이동한다.

stopstone committed:
feat: 앱 아이콘 변경

- dpi별 mipmap 설정

stopstone committed:
refactor: img_bg.png 이미지 dpi별 drawable 분리

stopstone committed:
refactor: 스플래시 화면 ViewModel 로직 수정

- navigation 함수를 viewModel에서 처리하던 기존 코드에서 Effect를 활용해 Screen에서 관리하는 코드로 변경

- 스플래시에서 uiState를 관리하지 않아 effect만 선언

stopstone committed:
feat: 온보딩 버튼 컴포넌트 추가

- ui/components의 NearBasicButton을 기반하여 만들었습니다.

- 현재 페이지가 마지막 페이지인지에 따라 버튼 텍스트를 "로그인/회원가입" 또는 "다음"으로 표시합니다.

- 버튼 클릭 시 `onNextClick` 콜백을 호출합니다.

stopstone committed:
feat: 온보딩 버튼 컴포넌트 추가

- ui/components의 NearBasicButton을 기반하여 만들었습니다.

- 현재 페이지가 마지막 페이지인지에 따라 버튼 텍스트를 "로그인/회원가입" 또는 "다음"으로 표시합니다.

- 버튼 클릭 시 `onNextClick` 콜백을 호출합니다.

stopstone committed:
feat: 온보딩 페이지 인디케이터 UI 구현

- 현재 페이지의 컬러를 BLUE01_5AA2E9로 설정
- 이외 컬리는 GRAY03_EBEBEB로 설정합니다.

stopstone committed:
feat: 온보딩 페이지 인디케이터 UI 구현

- 현재 페이지의 컬러를 BLUE01_5AA2E9로 설정
- 이외 컬리는 GRAY03_EBEBEB로 설정합니다.

stopstone committed:
feat: 온보딩 페이지 모델 구현

- 각각의 페이지의 구성요소를 담당하는 data class입니다.
- 추후 이미지를 추가하여 온보딩 페이지에서 이미지를 그리는 역할을 합니다.

stopstone committed:
feat: 온보딩 페이지 모델 구현

- 각각의 페이지의 구성요소를 담당하는 data class입니다.
- 추후 이미지를 추가하여 온보딩 페이지에서 이미지를 그리는 역할을 합니다.

stopstone committed:
feat: 온보딩 화면 구현

- 각 컴포넌트를 조합하여 온보딩 화면을 구현하였습니다.

- createAnnotatedText() 함수를 통해 \n 이후 텍스트를 강조합니다.

- stringResource를 통해 페이지 별 타이틀을 string.xml에서 가져옵니다.

stopstone committed:
feat: 온보딩 화면 구현

- 각 컴포넌트를 조합하여 온보딩 화면을 구현하였습니다.

- createAnnotatedText() 함수를 통해 \n 이후 텍스트를 강조합니다.

- stringResource를 통해 페이지 별 타이틀을 string.xml에서 가져옵니다.

stopstone committed:
feat: 온보딩 네비게이션 설정

stopstone committed:
feat: 온보딩 네비게이션 설정

stopstone committed:
feat: 온보딩 페이지 이미지 추가

stopstone committed:
feat: 온보딩 페이지 이미지 추가

stopstone committed:
feat: 온보딩 화면 이미지 추가

stopstone committed:
feat: 온보딩 화면 이미지 추가

stopstone committed:
feat: 온보딩 UI 이미지 영역 수정

stopstone committed:
feat: 온보딩 UI 이미지 영역 수정

stopstone committed:
feat: 온보딩 페이지 백그라운드 구성

stopstone committed:
feat: 온보딩 페이지 백그라운드 구성

stopstone committed:
feat: 온보딩 배경 추가

stopstone committed:
feat: 온보딩 배경 추가

stopstone committed:
refactor: 불필요한 패딩 제거

stopstone committed:
refactor: 불필요한 패딩 제거

stopstone committed:
feat: 온보딩 백그라운드 색상 변경

stopstone committed:
feat: 온보딩 백그라운드 색상 변경

stopstone committed:
refactor: OnboardingScreen UI 레이아웃 조정

OnboardingScreen의 UI 레이아웃을 수정하여 상태바와 네비게이션 바 영역을 고려하도록 변경했습니다.
`systemBarsPadding()` 대신 `WindowInsets`를 사용하여 직접 패딩을 계산하고 적용했습니다.

stopstone committed:
refactor: OnboardingScreen UI 레이아웃 조정

OnboardingScreen의 UI 레이아웃을 수정하여 상태바와 네비게이션 바 영역을 고려하도록 변경했습니다.
`systemBarsPadding()` 대신 `WindowInsets`를 사용하여 직접 패딩을 계산하고 적용했습니다.

stopstone committed:
refactor: AnnotatedString.Builder 확장 함수 추가

stopstone committed:
refactor: AnnotatedString.Builder 확장 함수 추가

stopstone committed:
feat: DataStore 분리 및 Qualifier 적용

stopstone committed:
feat: DataStore 분리 및 Qualifier 적용

stopstone committed:
feat: 온보딩 DataStore 추가

온보딩 완료 상태를 저장하고 조회하는 `OnboardingPreferences` 클래스를 추가했습니다.
이를 통해 앱 최초 실행 시 온보딩 화면 표시 여부를 결정할 수 있습니다.

stopstone committed:
feat: 온보딩 DataStore 추가

온보딩 완료 상태를 저장하고 조회하는 `OnboardingPreferences` 클래스를 추가했습니다.
이를 통해 앱 최초 실행 시 온보딩 화면 표시 여부를 결정할 수 있습니다.

stopstone committed:
feat: 온보딩 완료 상태 저장 로직 추가

stopstone committed:
feat: 온보딩 완료 상태 저장 로직 추가

stopstone committed:
feat: 온보딩 버튼 로딩 상태 추가

- 온보딩 버튼에 로딩 상태를 추가했습니다.
- 로딩 중일 때는 버튼이 비활성화됩니다.

stopstone committed:
feat: 온보딩 버튼 로딩 상태 추가

- 온보딩 버튼에 로딩 상태를 추가했습니다.
- 로딩 중일 때는 버튼이 비활성화됩니다.

stopstone committed:
feat: 온보딩 화면 UI 상태 및 사이드 이펙트 처리

- OnboardingViewModel에 UiState 및 Effect 추가
- OnboardingScreen에서 UiState 관찰 및 Effect 처리 로직 구현
- 온보딩 완료 시 로딩 상태 표시 및 에러 처리 추가

stopstone committed:
feat: 온보딩 화면 UI 상태 및 사이드 이펙트 처리

- OnboardingViewModel에 UiState 및 Effect 추가
- OnboardingScreen에서 UiState 관찰 및 Effect 처리 로직 구현
- 온보딩 완료 시 로딩 상태 표시 및 에러 처리 추가

stopstone committed:
feat: 온보딩 버튼 텍스트 stirng.xml 리소스 추가

stopstone committed:
feat: 온보딩 버튼 텍스트 stirng.xml 리소스 추가

stopstone committed:
feat: 온보딩 레포지토리 추가 및 ViewModel 연동

- 온보딩 완료 여부 저장 및 조회 기능을 위한 `OnBoardingRepository` 인터페이스와 `OnBoardingRepositoryImpl` 구현체를 추가했습니다.
- `OnboardingViewModel`이 `OnboardingPreferences` 대신 `OnBoardingRepository`를 사용하도록 수정했습니다.
- `RepositoryModule`에 `OnBoardingRepository` 관련 의존성 주입 설정을 추가했습니다.

stopstone committed:
feat: 온보딩 레포지토리 추가 및 ViewModel 연동

- 온보딩 완료 여부 저장 및 조회 기능을 위한 `OnBoardingRepository` 인터페이스와 `OnBoardingRepositoryImpl` 구현체를 추가했습니다.
- `OnboardingViewModel`이 `OnboardingPreferences` 대신 `OnBoardingRepository`를 사용하도록 수정했습니다.
- `RepositoryModule`에 `OnBoardingRepository` 관련 의존성 주입 설정을 추가했습니다.

stopstone committed:
refactor: 온보딩 페이지 데이터 - remember로 성능 최적화

stopstone committed:
refactor: 온보딩 페이지 데이터 - remember로 성능 최적화

stopstone committed:
refactor: 온보딩 Preferences 키 상수화

stopstone committed:
refactor: 온보딩 Preferences 키 상수화

stopstone committed:
refactor: 불필요한 코드 제거

stopstone committed:
refactor: 불필요한 코드 제거

stopstone committed:
refactor: 온보딩 페이지 데이터 모델 변경

- `title` 필드를 `String`에서 `@StringRes` 어노테이션을 사용하는 `Int` 타입의 `titleResId`로 변경했습니다.
- 이에 따라 `OnboardingScreen`에서 `title`을 참조하는 부분을 `stringResource`를 사용하도록 수정했습니다.

stopstone committed:
refactor: 온보딩 페이지 데이터 모델 변경

- `title` 필드를 `String`에서 `@StringRes` 어노테이션을 사용하는 `Int` 타입의 `titleResId`로 변경했습니다.
- 이에 따라 `OnboardingScreen`에서 `title`을 참조하는 부분을 `stringResource`를 사용하도록 수정했습니다.

stopstone committed:
refactor: 디자인 가이드에 맞게 텍스트 수정

stopstone committed:
refactor: 디자인 가이드에 맞게 텍스트 수정

stopstone committed:
feat: Color.kt 온보딩 배경 장식 색상 추가

stopstone committed:
feat: Color.kt 온보딩 배경 장식 색상 추가

stopstone committed:
refactor: 불필요한 import문 제거

stopstone committed:
refactor: 불필요한 import문 제거

rhkrwngud445 committed:
feat: 연락처 데이터 호출 로직 작업

rhkrwngud445 committed:
feat: 연락처 데이터 호출 로직 작업

rhkrwngud445 committed:
feat: 연락처 권한 추가

rhkrwngud445 committed:
feat: 연락처 권한 추가

stopstone committed:
feat: 마이페이지 구성에 필요한 리소스 추가

stopstone committed:
feat: LoginType enum 추가

- ProviderType에 따라 텍스트와 아이콘을 가진다.

stopstone committed:
feat: 마이프로필 컴포넌트 추가

- NearLogoutButton: 로그아웃 버튼
- NearServiceInfoRow: 서비스 정보 행
- NearSocialLoginBadge: 소셜 로그인 뱃지
- NearSwitch: 토글 스위치

stopstone committed:
feat: 내 정보 화면 UI 및 ViewModel 구현

stopstone committed:
refactor: LoginType model 패키지로 이동

stopstone committed:
feat: 내 프로필 화면 navigation 구현

- 홈에서 MY 클릭 시 마이페이지 이동

stopstone committed:
feat: Glide Compose 라이브러리 추가

stopstone committed:
feat: 회원 정보 API 연동

rhkrwngud445 committed:
feat: 연락처 데이처 호출 로직 적용

stopstone committed:
feat: MyProfile 에러 핸들링 추가

stopstone committed:
feat: 프로필 이미지 로딩 구현

stopstone committed:
feat: 멤버 정보 API 연동

- MemberRepository 인터페이스 및 구현체 추가
- getMyInfo API 호출 로직 구현
- RepositoryModule에 MemberRepository 바인딩

stopstone committed:
feat: 내 정보 화면 및 API 연동

- 내 정보 화면 UI 구현
- 내 정보 API 연동 및 ViewModel, UiState, UiEvent 정의
- 사용자 닉네임, 로그인 타입 표시
- 알림 설정, 로그아웃, 회원탈퇴 기능 UI 구성

stopstone committed:
feat: 마이프로필 뒤로가기 기능 구현

- MyProfileViewModel에 UI 이벤트 채널 및 관련 로직 추가
- MyProfileScreen에서 뒤로가기 버튼 클릭 시 내비게이션 이벤트 처리
- MyProfileScreen의 NearTopAppbar에 뒤로가기 버튼 핸들러 연결

stopstone committed:
refactor: 마이프로필 정보에 매퍼 적용 및 UI 연결

- MemberInfo를 MyProfileInfo로 변환하는 매퍼 추가
- MyProfileViewModel에서 MemberInfo 대신 MyProfileInfo 사용하도록 수정
- MyProfileScreen에서 MyProfileInfo를 사용하여 UI 업데이트

stopstone committed:
feat: 마이프로필 로그아웃 기능 구현

- MyProfileViewModel에 authRepository를 주입하여 로그아웃 로직을 추가했습니다.
- 로그아웃 성공 시 로그인 화면으로 이동하도록 NavigationMyProfile, MyProfileScreen, NearNavHost를 수정했습니다.
- MyProfileUiEvent에 Logout 이벤트를 추가했습니다.

stopstone committed:
refactor: 로딩 화면 범위 수정

stopstone committed:
refactor: MyProfileScreen 부분별 코드 분리

- MyProfileInfoSection(): 프로플 정보를 나타내는 부분
- MyProfileGeneralSection(): 연결계정, 알림설정 및 계쩡 관련 부분
- MyProfileInfoSection(): Near앱 서비스 고지 관련 부분

rhkrwngud445 committed:
feat: 연락처 기념일, 라벨 등 추가정보 적용

rhkrwngud445 committed:
feat: 연락처 기념일, 라벨 등 추가정보 적용

rhkrwngud445 committed:
feat: 연락처 화면 파일 추가

rhkrwngud445 committed:
feat: 연락처 화면 파일 추가

stopstone committed:
feat: 회원탈퇴 화면 이동 로직 구현

stopstone committed:
feat: 회원탈퇴 API 연동

stopstone committed:
feat: 회원탈퇴 사유 모델 추가

- WithdrawReason enum 클래스 추가
- WithdrawReasonItem 데이터 클래스 추가

stopstone committed:
feat: 회원탈퇴 화면 TopAppBar 구현

- 백 버튼 없이 취소 버튼만 있는 `WithdrawTopAppBar` Composable을 추가했습니다.
- `title`과 `onCancelClick`을 파라미터로 받아 화면 제목과 취소 버튼 클릭 시 동작을 설정할 수 있습니다.

stopstone committed:
feat: 회원탈퇴 화면 및 기능 구현

- 회원탈퇴 화면 UI를 구성했습니다.
- 회원탈퇴 이유 선택 및 기타 이유 입력 기능을 추가했습니다.
- 회원탈퇴 API 연동 및 로그아웃 후 로그인 화면 이동 로직을 구현했습니다.
- 회원탈퇴 화면 상단바에 사용될 취소 아이콘(ic_32_cancel)을 추가했습니다.

stopstone committed:
feat: NearOutlinedTextField 컴포넌트 추가

- 커스텀 `OutlinedTextField` 구현
- 글자 수 표시 기능 추가
- 다양한 상태에 대한 Preview 추가
- TODO 키보드와 텍스트필드 사이 패딩 필요

stopstone committed:
refactor: NearOutlinedTextField 키보드 상호작용 개선

- 키보드 위에 텍스트필드를 배치
- 글자가 입력될 때 텍스트필드 사이즈를 반영하여 키보드 위에 배치
- 패딩값 동적으로 계산하여 반영

rhkrwngud445 committed:
feat: 연락처 상단바 추가

rhkrwngud445 committed:
feat: Near 검색 TextField 추가

rhkrwngud445 committed:
feat: Near 검색 TextField 추가

rhkrwngud445 committed:
feat: 화면 검색 컴포넌트 추가

rhkrwngud445 committed:
feat: 화면 검색 컴포넌트 추가

rhkrwngud445 committed:
feat: 연락처 리스트 화면 구성

rhkrwngud445 committed:
feat: 연락처 리스트 화면 구성

rhkrwngud445 committed:
fix: 연락처 리스트 마진 수정

rhkrwngud445 committed:
fix: 연락처 리스트 마진 수정

rhkrwngud445 committed:
feat: 연락처 로딩 및 체크박스 인터렉션 적용

rhkrwngud445 committed:
feat: 연락처 로딩 및 체크박스 인터렉션 적용

rhkrwngud445 committed:
fix: 쌍자음도 기본 자음에 포함되도록 수정

rhkrwngud445 committed:
fix: 쌍자음도 기본 자음에 포함되도록 수정

rhkrwngud445 committed:
feat: navigation 화면 정의

rhkrwngud445 committed:
feat: 검색 인터렉션 구현

rhkrwngud445 committed:
feat: 한글, 영어, 특수문자 순으로 연락처 정렬

rhkrwngud445 committed:
feat: 한글, 영어, 특수문자 순으로 연락처 정렬

rhkrwngud445 committed:
fix: 이름만 검색되도록 수정

rhkrwngud445 committed:
fix: 이름만 검색되도록 수정

rhkrwngud445 committed:
feat: 우측 자음 인덱스바 추가

rhkrwngud445 committed:
feat: 우측 자음 인덱스바 추가

rhkrwngud445 committed:
feat: permission 요청 컴포저블 추가

rhkrwngud445 committed:
feat: permission 요청 컴포저블 추가

stopstone committed:
feat: 텍스트필드 에러 상태 및 관련 기능 추가

- 텍스트필드에 `isError` 상태를 추가하고, 에러 발생 시 에러 메시지를 표시하도록 구현했습니다.
- 에러가 발생하면 해당 뷰로 자동 스크롤되는 기능을 추가했습니다.
- `focusRequester`를 외부에서 주입받아 포커스를 제어할 수 있도록 수정했습니다.
- 한 번이라도 입력이 시작되면 글자 수가 항상 표시되도록 로직을 변경했습니다.

stopstone committed:
feat: 회원탈퇴 화면 UI 및 로직 개선

- `WithdrawUiState`에 탈퇴하기 버튼 활성화 및 기타 사유 입력 로직을 관리하는 상태 프로퍼티를 추가했습니다.
- '기타' 사유 선택 시, 키보드가 먼저 올라오도록 포커스를 요청하고 텍스트 필드를 초기화하는 로직을 추가했습니다.
- 기타 사유를 입력할 수 있는 `NearOutlinedTextField`를 추가하고, 선택 여부에 따라 활성화 상태를 제어합니다.

stopstone committed:
style: 회원탈퇴 화면 앱바 패딩 수정

stopstone committed:
refactor: 의견 입력 에러 메시지 수정

stopstone committed:
feat: 회원탈퇴 화면 뒤로가기 추가

stopstone committed:
feat: 회원탈퇴 화면에 닉네임 전달 로직 추가

- saveStateHandle로 닉네임 값을 넘겨받아 닉네임을 UI에 보여주기

stopstone committed:
chore: 탈퇴 화면 하단 패딩 수정

stopstone committed:
chore: NearServiceInfoRow 클릭 범위 수정 및 리플효과 제거

stopstone committed:
refactor: WithdrawTopAppBar → NearCancelTopAppBar로 이름 변경 및 이동

- 공통 컴포넌트로 활용하기 위해 ui/componenets로 이동

stopstone committed:
chore: 미사용 import 제거

stopstone committed:
refactor: Modifier 파라미터 적용

stopstone committed:
chore: build.gradle.kts 약관 URL 정보 추가

stopstone committed:
feat: 약관 및 정책 웹뷰 화면 연동

- 약관 및 정책의 제목과 URL을 관리하는 `TermsType` enum을 추가했습니다.
- 마이프로필 화면에서 약관 및 정책 항목을 클릭하면 해당 URL을 웹뷰로 여는 기능을 추가했습니다.
- 웹뷰를 표시하기 위한 `WebView` composable과 관련 네비게이션 로직을 추가했습니다.

stopstone committed:
refactor: 약관 및 정책 목록 표시 로직 개선

- TermsType enum class를 활용하여 NearServiceInfoRow를 반복하여 호출하였습니다.

stopstone committed:
refactor: 약관 페이지 title 형식 수정

- title 뒤에 "상세" 문자열 추가

stopstone committed:
feat: MY 페이지, 탈퇴하기 화면 string.xml 리소스 적용

- MY 페이지, 탈퇴하기 화면에서 사용할 스트링 리소스를 string.xml로 관리합니다.
- 하드코딩된 문자열을 `stringResource`를 사용하여 스트링 리소스로 대체했습니다.
- `LoginType`, `TermsType`, `WithdrawReason` enum 클래스가 스트링 리소스 ID를 갖도록 수정했습니다.

stopstone committed:
remove: 회원 탈퇴 사유 모델 제거

stopstone committed:
refactor: LoginType의 프로퍼티명 수정

stopstone committed:
Merge branch 'dev' into feat/splash-auto-login

stopstone committed:
feat: 스플래시 화면 관련 파일 삭제

stopstone committed:
feat: 스플래시 화면 라이브러리 추가

- Android 12 이상 버전과의 호환성을 위해 `core-splashscreen` 라이브러리를 추가했습니다.

stopstone committed:
feat: themes.xml에서 스플래시 화면 구현

- SplashScreen API를 사용하여 스플래시 화면을 추가했습니다.
- MainActivity에 스플래시 테마를 적용했습니다.
- values/themes.xml에 스플래시 화면 스타일을 정의했습니다.

stopstone committed:
feat: 스플래시 로고 이미지 추가

stopstone committed:
feat: 스플래시 화면 API 적용 및 로그인 상태에 따른 화면 분기 처리

- 앱 시작 시 로그인 상태를 확인하고, 그 결과에 따라 초기 화면(로그인 또는 홈)으로 이동하도록 로직을 구현했습니다.
- `MainViewModel`을 추가하여 로그인 상태 확인 및 UI 상태를 관리합니다.
- `MainActivity`에서 `installSplashScreen`을 사용하여 API 스플래시를 적용하고, 로그인 상태 확인이 완료될 때까지 스플래시 화면이 유지되도록 수정했습니다.

stopstone committed:
chore: 내 정보 화면 배경색 추가

stopstone committed:
Merge pull request #26 from near-Contact-Reminder/feat/splash-auto-login

[Feature] 스플래시 화면 구성 및 자동 로그인 구현

stopstone committed:
feat: 앱 시작 시 온보딩 및 로그인 상태에 따른 화면 분기 처리

- MainViewModel에 onBoardingRepository를 주입하여 온보딩 완료 여부를 확인합니다.
- MainActivity, NearApp, NearNavHost에서 로그인 상태(isLoggedIn) 대신 시작 화면(startDestination)을 사용하도록 수정했습니다.
- MainUiState에 isOnboardingCompleted와 startDestination 필드를 추가했습니다.
- 앱 시작 시 온보딩 완료 여부와 로그인 상태를 확인하여 적절한 화면(온보딩, 홈, 로그인)으로 이동하도록 MainViewModel의 checkAppStatus 로직을 수정했습니다.
- TODO  각 루트의 타입이 다르기에 Any로 설정하였습니다. 나중에 하나의 루트로 두면 좋을 것 같습니다

stopstone committed:
Merge remote-tracking branch 'origin/feat/onboarding-pages' into feat/onboarding-pages

# Conflicts:
#	Near/app/src/main/java/com/alarmy/near/presentation/feature/main/NearNavHost.kt

stopstone committed:
refactor: 온보딩 관련 중복 문자열 제거

- values/strings.xml에서 중복된 온보딩 관련 문자열 리소스를 삭제했습니다.

stopstone committed:
Merge branch 'dev' into feat/my-profile

stopstone committed:
refactor: 인증 헤더 삭제

- `MemberApiService`의 `getMyInfo`, `withdraw` 함수에서 Authorization 헤더 파라미터 삭제
- `MemberRepositoryImpl`에서 `getMyInfo`, `withdraw` 호출 시 Authorization 헤더 전달 로직 삭제

stopstone committed:
refactor: MyProfileUiState에서 사용하지 않는 error 필드 제거

stopstone committed:
refactor: RouteWebView에서 toRoute 사용하도록 수정

- 기존에 arguments에서 직접 값을 가져오던 방식에서 `toRoute<RouteWebView>()`를 사용하도록 변경했습니다.

stopstone committed:
feat: WebView 뒤로가기 기능 추가

- WebViewFrame 컴포저블에 BackHandler를 추가하여 WebView 내에서 뒤로가기 기능을 구현했습니다.
- WebView의 canGoBack 상태를 추적하여 뒤로가기 가능 여부를 결정합니다.
- 페이지 로드 완료 시 및 WebView 업데이트 시 canGoBack 상태를 업데이트합니다.

stopstone committed:
refactor: Glide의 contentScale 옵션 제거

- GlideImage의 contentScale 파라미터가 이미 존재하므로 중복되는 RequestOptions의 contentScale 관련 코드를 삭제했습니다.

stopstone committed:
fix: `NearOutlinedTextField` 에러 상태 반영 오류 수정

- `OutlinedTextFieldDefaults.Container`의 `isError` 파라미터에 `isError` 상태를 직접 전달하도록 수정하여, 텍스트 필드의 에러 상태가 올바르게 반영되도록 했습니다.

stopstone committed:
refactor: 회원탈퇴 화면 에러 처리 방식 변경

- WithdrawViewModel에서 errorMessage를 직접 관리하는 대신 WithdrawUiEvent.ShowError 이벤트를 발생시켜 UI에 에러를 전달하도록 수정했습니다.
- WithdrawScreen에서 ShowError 이벤트를 수신하여 스낵바를 통해 에러 메시지를 표시하도록 변경했습니다.
- WithdrawUiState에서 errorMessage 필드를 제거했습니다.
- NavigationMyProfile의 WithdrawRoute에 onShowErrorSnackBar 파라미터를 추가하여 에러 발생 시 스낵바를 표시할 수 있도록 했습니다.

stopstone committed:
chore: ci 워크플로우에 개인정보처리방침 URL 추가

stopstone committed:
Merge pull request #29 from near-Contact-Reminder/feat/onboarding-pages

[Feature] 앱 초기 튜토리얼 화면

stopstone commented on Near/app/src/main/java/com/alarmy/near/presentation/feature/contact/ContactScreen.kt:
```diff
@@ -52,7 +57,13 @@ fun ContactScreen(
+        NearSearchTextField(
+            placeHolderText = stringResource(R.string.context_search_placeholder),
+            modifier = Modifier.padding(horizontal = 20.dp),
+            value = "",
```
value가 기본파라미터가 아니여서 호출할 때 빈 값을 명시해주신 것 같은데,
텍스트필드가 처음에는 값이 없는 경우도 있어서 value를 기본 파라미터로 주어도 좋을 것 같습니다!

rhkrwngud445 replied:
앗 리뷰주실 때 변경이 반영이 안되었네요! viewModel의 searchQuery를 파라미터로 사용했습니다!

stopstone commented on Near/app/src/main/java/com/alarmy/near/presentation/feature/contact/ContactScreen.kt:
```diff
@@ -17,36 +17,73 @@ import androidx.compose.foundation.layout.padding
+    when (uiState) {
+        is ContactUiState.Loading -> {
+            Box(modifier = Modifier.fillMaxSize()) {
+                CircularProgressIndicator(
```
로딩 인디케이터 색상이 기본적으로 머터리얼의 primary 색상이 들어가는 것으로 알고 있습니다..!
색상을 명시해줘도 좋을 것 같습니다~!

rhkrwngud445 replied:
오 아무래도 브랜드 컬러로 통일하는 것이 좋을 것 같네요! BLUE01로 적용했습니다!

stopstone commented on Near/app/src/main/java/com/alarmy/near/presentation/feature/main/NearNavHost.kt:
```diff
@@ -33,7 +37,7 @@ internal fun NearNavHost(
         modifier = modifier,
         navController = navController,
-        startDestination = RouteHome,
+        startDestination = RouteContact,
```
이 부분은 튜토리얼 관련 브랜치에서 작업된 것으로 반영하면 좋을 것 같습니다!

stopstone commented on Near/app/src/main/java/com/alarmy/near/presentation/feature/contact/ContactViewModel.kt:
```diff
@@ -57,15 +57,15 @@ class ContactViewModel
         }
 
+        // 초성을 추출하는 함수
         fun getInitial(name: String): String {
```
이런식으로 초성을 추철할 수 있군요..! 배워갑니다 :)

stopstone commented on Near/app/src/main/java/com/alarmy/near/presentation/feature/contact/ContactScreen.kt:
```diff
@@ -64,6 +77,82 @@ fun ContactScreen(
+    val sectionedContacts = groupedContacts.toList()
+    LazyColumn(modifier = Modifier.fillMaxSize()) {
+        sectionedContacts.forEach { (initial, contacts) ->
+            stickyHeader {
```
아 stichyHeader라는 함수가 있군요...!

stopstone commented on Near/app/src/main/java/com/alarmy/near/presentation/feature/contact/ContactScreen.kt:
```diff
@@ -64,6 +77,82 @@ fun ContactScreen(
+    val sectionedContacts = groupedContacts.toList()
+    LazyColumn(modifier = Modifier.fillMaxSize()) {
+        sectionedContacts.forEach { (initial, contacts) ->
+            stickyHeader {
```
stickyHeader가 스크롤 되도 상단에 고정되는 영역이라고 보았는데,
sectionedContacts.forEach { (initial, contacts) ->
내부에 있으면 고정된 요소를 계속 반복해서 호출하게 되나요?

혹시, stickyHeader 가 이러한 리컴포지션을 방지해주나요?

rhkrwngud445 replied:
넵 맞습니다! 초성 타이틀 바를 상단에 고정하는 역할을 하고 있어요!
LazyColumn이 stickyHeader를 포함하여 레이아웃을 정보를 재활용하여 리컴포지션을 방지해줍니다!
stickHeader는 고정되는 뷰일뿐 따로 성능상 이점은 없는 것으로 알고 있어요!

stopstone commented on Near/app/src/main/java/com/alarmy/near/presentation/feature/contact/ContactScreen.kt:
```diff
@@ -87,6 +89,17 @@ fun ContactScreen(
             ContactList(
                 groupedContacts = groupedContacts,
             )
+            NearSolidTypeButton(
```
우측 인덱스바와 버튼의 여유가 부족해보여서 패딩을 살짝 주면 어떨까요?
디자인 요구사항에서는 12dp 정도 여유가 있는 것으로 보입니다
 
<img width="648" height="1447" alt="image" src="https://github.com/user-attachments/assets/8b37faa0-2e22-4ef9-9394-85b6aed01818" />

rhkrwngud445 replied:
말씀주신 패딩은 버튼 상단 패딩을 말씀주시는 것 맞을까요?
패딩 방향이 조금 헷갈리네요🥲

stopstone replied:
앗! 하단 패딩이였습니다! 
제가 명확하게 말씀드리지 못했네요 ㅠㅠ

rhkrwngud445 replied:
혹시 이 주황색 영역 맞을까요..? 제가 공간지각 능력이..🥲
<img width="310" height="651" alt="스크린샷 2025-09-23 오후 9 33 21" src="https://github.com/user-attachments/assets/9668aa7e-9bc2-426c-af15-82f44fc5f6ac" />

stopstone replied:
<img width="511" height="1154" alt="image" src="https://github.com/user-attachments/assets/d61f98a3-0e0c-4c6c-9a61-d9cb6881dd55" />


이 영역입니다..! 
우측 인덱스 바(#) 이랑 버튼이 붙어 있는 것 같아 말씀 드렸습니다!
주형님 애뮬레이터에서는 여유가 상당히 많군요~ :)

앗 참고로 Z플립 5로 실행하였습니다!!

rhkrwngud445 replied:
이미지 설명 감사합니다! 확실히 붙어 있긴 하네요..
상단바 마진이 잘못 적용되어 있었어요! 수정했습니다!

stopstone commented:
리뷰가 늦었습니다 ㅠ_ㅠ
권한처리를 제가 해야될 것 같아서 우선 프리뷰만 보고 리뷰하였습니다!
제가 하기에는 상당히 어려운 작업이였을 것 같은데 보고 배웠습니다 
고생하셨어요!!

stopstone reviewed (commented):
리뷰가 늦었습니다 ㅠ_ㅠ
권한처리를 제가 해야될 것 같아서 우선 프리뷰만 보고 리뷰하였습니다!
제가 하기에는 상당히 어려운 작업이였을 것 같은데 보고 배웠습니다
고생하셨어요!!

stopstone committed:
refactor: Glide에서 Coil로 이미지 로딩 라이브러리 변경

- 기존 Glide를 사용하던 ImageLoader를 Coil을 사용하도록 수정했습니다.
- `AsyncImage`를 사용하여 이미지를 비동기적으로 로드하고, `ImageRequest.Builder`를 통해 crossfade 효과를 적용했습니다.
- crossfade는 논의 후붎필요하면 제거하겠습니다!
- 관련 의존성을 `libs.versions.toml` 및 `app/build.gradle.kts`에서 Glide에서 Coil로 변경하였습니다.

stopstone committed:
refactor: API 응답 타입 수정

- `MemberApiService`의 `getMyInfo` 함수의 반환 타입을 `Response<MemberInfo>`에서 `MemberInfo`로 변경했습니다.
- `withdraw` 함수의 반환 타입을 `Response<Unit>`에서 `Unit`으로 변경했습니다.

stopstone committed:
feat: flow를 간소화 한 apiCallFlow 추가
- `apiCallFlow` 함수를 추가하여 Repository에서 API 호출 시 예외 처리를 단순화했습니다.
- apiCallFlow를 활용하면 emit을 생략할 수 있습니다.

stopstone committed:
refactor: API 호출 Flow 적용

- 불필요한 runCatching을 제거하였습니다.
- `apiCallFlow` 함수를 추가하여 Repository에서 API 호출 시 예외 처리를 단순화했습니다.
- `MemberRepositoryImpl`의 `getMyInfo` 및 `withdraw` 함수에 `apiCallFlow`를 적용하여 중복 코드를 제거하고 가독성을 향상했습니다.

stopstone committed:
refactor: 회원 탈퇴 API 에러 핸들링 수정

- 회원 탈퇴 API 호출 시 `catch`를 사용하여 에러를 처리하도록 수정했습니다.

stopstone committed:
refactor: Member 관련 데이터 모델 네이밍 및 매퍼 구조 변경

- `MemberInfoEntity` 및 `WithdrawRequestEntity`를 추가하여 Data Layer의 API 요청/응답 모델을 정의했습니다.
- 기존 Model Layer의 `MemberInfo`, `WithdrawRequest`를 Data Layer와 Presentation Layer 사이의 중간 모델로 사용하도록 수정했습니다.
- `MemberMapper`를 개편하여 각 Layer 간 모델 변환 로직을 명확히 분리했습니다.
    - Data Layer Entity <-> Model Layer 변환 함수 추가
    - Model Layer <-> UI Layer 변환 함수명 변경 (`toMyProfileInfo` -> `toMyProfileInfoUIModel`)
- `MemberRepositoryImpl`에서 API 호출 시 Data Layer Entity를 사용하도록 수정했습니다.
- `MyProfileViewModel`에서 UI 모델명 변경사항을 반영했습니다. (`MyProfileInfo` -> `MyProfileInfoUIModel`)

stopstone committed:
refactor: `WithdrawRequest` 모델의 위치 변경 및 관련 로직 수정

- `WithdrawRequest` 모델을 `model/member` 패키지에서 `network/request` 패키지로 이동했습니다.
- `MemberRepository`의 `withdraw` 함수 시그니처를 변경했습니다.
    - 기존: `fun withdraw(request: WithdrawRequest): Flow<Unit>`
    - 변경: `fun withdraw(reason: WithdrawReason, customReason: String? = null): Flow<Unit>`

stopstone committed:
fix: 중복된 NavGraph 정의 제거
- 머지 충돌로 인하여 중복된 NavGraph를 제거합니다.

stopstone committed:
refactor: `MyProfileInfo`를 `MyProfileInfoUIModel`로 이름 변경

stopstone committed:
refactor: 불필요한 import 제거

stopstone committed:
Merge branch 'feat/my-profile' into dev

# Conflicts:
#	Near/app/src/main/java/com/alarmy/near/data/di/RepositoryModule.kt
#	Near/app/src/main/java/com/alarmy/near/presentation/feature/main/NearNavHost.kt

stopstone committed:
refactor: `NearNavHost` 시작 화면 `startDestination`으로 통합

- `NearNavHost`의 시작 화면을 결정하는 `isLoggedIn` 파라미터를 제거하고, `startDestination` 파라미터로 통합하여 외부에서 시작 화면을 직접 지정하도록 변경했습니다.

rhkrwngud445 committed:
feat: 로딩 인디케이터 색상 변경

rhkrwngud445 committed:
feat: 로딩 인디케이터 색상 변경

stopstone committed:
Merge pull request #33 from near-Contact-Reminder/feat/my-profile

[FEAT] 마이 프로필 및 회원 탈퇴

rhkrwngud445 committed:
fix: 인덱스바 상단 마진 축소

rhkrwngud445 committed:
fix: 인덱스바 상단 마진 축소

rhkrwngud445 committed:
feat: 연락처 데이처 호출 로직 적용

# Conflicts:
#	Near/app/src/main/java/com/alarmy/near/data/di/RepositoryModule.kt

rhkrwngud445 committed:
feat: 연락처 상단바 추가

# Conflicts:
#	Near/app/src/main/res/values/strings.xml

rhkrwngud445 committed:
feat: navigation 화면 정의

# Conflicts:
#	Near/app/src/main/java/com/alarmy/near/presentation/feature/main/NearNavHost.kt

rhkrwngud445 committed:
feat: 검색 인터렉션 구현

# Conflicts:
#	Near/app/src/main/java/com/alarmy/near/presentation/feature/main/NearNavHost.kt

rhkrwngud445 committed:
Merge remote-tracking branch 'origin/feat/#27-contact_picker_data_layer' into feat/#27-contact_picker_data_layer

# Conflicts:
#	Near/app/src/main/java/com/alarmy/near/data/di/RepositoryModule.kt
#	Near/app/src/main/java/com/alarmy/near/presentation/feature/main/NearNavHost.kt
#	Near/app/src/main/res/values/strings.xml
